### ISSUE TITLE
No audio detected

### ISSUE THREAD

**Issue by KpauMupa** [SeaDve/Mousai#33]
![image](https://user-images.githubusercontent.com/76217453/132508398-d662340a-ff77-4cf7-9ba6-70808a4112f9.png)

Arch linux with most recent updates. I even installed a GNOME DE to see if Mousai can catch the sound there - still this issue. 
Maybe I need some kind of awkward sound mixer? 
I have those sound-related packages installed: alsa-utils, pulseaudio, pulseaudio-alsa and pavucontrol.

**Comment by SeaDve** [comment]
try changing preferred audio source in hamburger menu

**Comment by KpauMupa** [comment]
> try changing preferred audio source in hamburger menu

Of course I tried! It's still nothing =( 
I don't have a mic, btw - maybe this is an issue?

**Comment by SeaDve** [comment]
it gets the default audio source as stated in `pactl info`. So i think it should not matter

if you have gnome control center installed you can check the device there

**Comment by KpauMupa** [comment]
> it gets the default audio source as stated in `pactl info`. So i think it should not matter
> 
> if you have gnome control center installed you can check the device there

![image](https://user-images.githubusercontent.com/76217453/132510790-71fd9a1b-3c59-400c-b0ae-bb8e13c043ef.png)
Everything's fine, sound's showing

**Comment by KpauMupa** [comment]
> it gets the default audio source as stated in `pactl info`. So i think it should not matter
> 
> if you have gnome control center installed you can check the device there

I reinstalled mousai a couple of times, rebooted a few times and now it's working, even though I never touched anything in my sound settings and never after installed any sound-related packages

**Comment by SeaDve** [comment]
that's really weird

**Comment by KpauMupa** [comment]
> that's really weird, may you show the output of `pactl info`? it may be helpful

Will it be an issue that it's in a different language? I can change my local and reboot, if it *is* an issue

Строка сервера: /run/user/1000/pulse/native
Версия протокола библиотеки: 35
Версия протокола сервера: 35
Выполняется локально: да
Номер клиента: 9
Размер блока памяти: 65472
Имя пользователя: anonymous
Имя хоста: MyCompooter
Имя сервера: pulseaudio
Версия сервера: 15.0
Спецификация отсчётов по умолчанию: s16le 2-канальный 4410
Схема каналов по умолчанию: front-left,front-right
Аудиоприёмник по умолчанию: alsa_output.pci-0000_29_00.6.analog-stereo
Источник по умолчанию: alsa_output.pci-0000_29_00.6.analog-stereo.monitor
Cookie: 88e2:4950

**Comment by KpauMupa** [comment]
> that's really weird, may you show the output of `pactl info`? it may be helpful

And by the way, after I completely removed GNOME DE from my PC (sudo pacman  -Rsc gnome), reinstalled mousai (dependencies broke) and rebooted my PC - Mousai doesn't work again XD

**Comment by KpauMupa** [comment]
> that's really weird, may you show the output of `pactl info`? it may be helpful

What do I need to install for Mousai to work in my DWM? There's a really big package list in a "gnome" pack, I don't know what exactly I have to install

**Comment by KpauMupa** [comment]
> that's really weird, may you show the output of `pactl info`? it may be helpful

So I looked thru the list of packages of GNOME and installed `soundtouch`, rebooted - now Mousai works... 

I'm fairly new to linux and I don't know what this package does, but it worked out for me.
BTW, here's my `pactl info` *after*  I installed soundtouch: 
``` anonymous  MyCompooter  ~  $  pactl info                                            15:57
Server String: /run/user/1000/pulse/native
Library Protocol Version: 35
Server Protocol Version: 35
Is Local: yes
Client Index: 6
Tile Size: 65472
User Name: anonymous
Host Name: MyCompooter
Server Name: pulseaudio
Server Version: 15.0
Default Sample Specification: s16le 2ch 44100Hz
Default Channel Map: front-left,front-right
Default Sink: alsa_output.pci-0000_29_00.6.analog-stereo
Default Source: alsa_output.pci-0000_29_00.6.analog-stereo.monitor
Cookie: 9aa9:ebc6```

**Comment by KpauMupa** [comment]
Guess I have to close this issue...

**Comment by KpauMupa** [comment]
No... This issue is here again, after a reboot. Is this because Arch is weird or because I am? =)

**Comment by SeaDve** [comment]
im not sure either whats happening :D

**Comment by SeaDve** [comment]
since pactl info is working, i think nothing else

**Comment by KpauMupa** [comment]
> since pactl info is working, i think nothing else.
> 
> terminal output, if any, while reproducing the issue may be helpful too, as it may be a missing plugin

So I changed my locale so it's readable for you, restarted and... And Mousai worked!

After the previous reboot I INSTALLED NOTHING and DID NOTHING else. 

Maybe it's the issue with locales???

**Comment by KpauMupa** [comment]
And again `pactl info`:
``` anonymous  MyCompooter  ~  $  pactl info                                            16:30
Server String: /run/user/1000/pulse/native
Library Protocol Version: 35
Server Protocol Version: 35
Is Local: yes
Client Index: 2
Tile Size: 65472
User Name: anonymous
Host Name: MyCompooter
Server Name: pulseaudio
Server Version: 15.0
Default Sample Specification: s16le 2ch 44100Hz
Default Channel Map: front-left,front-right
Default Sink: alsa_output.pci-0000_29_00.6.analog-stereo
Default Source: alsa_output.pci-0000_29_00.6.analog-stereo.monitor
Cookie: 8636:7b9d```

**Comment by SeaDve** [comment]
ah yess i think so, ironically, lmao

**Comment by SeaDve** [comment]
to validate, it would be helpful to check the output of mousai when run on terminal

**Comment by KpauMupa** [comment]
> to validate, it would be helpful to check the output of mousai when run on terminal
So I rebooted with my default ru_RU.UTF-8 locale again, and my theory is working - it's the issue with locales. 

And here is the terminal output of Mousai with my ru_RU.UTF-8 locale you asked for:
``` anonymous  MyCompooter  ~  $  mousai                                          2.23s 16:33

(io.github.seadve.Mousai:2326): GStreamer-CRITICAL **: 16:34:47.452: _gst_util_uint64_scale: assertion 'denom != 0' failed

(io.github.seadve.Mousai:2326): GStreamer-CRITICAL **: 16:34:47.452: gst_segment_to_running_time: assertion 'segment->format == format' failed

(io.github.seadve.Mousai:2326): GStreamer-CRITICAL **: 16:34:47.452: gst_segment_to_stream_time: assertion 'segment->format == format' failed
Error: gst-resource-error-quark: Failed to connect stream: Недопустимый параметр (1) ../gst-plugins-good/ext/pulse/pulsesrc.c(1619): gst_pulsesrc_prepare (): /GstPipeline:pipeline0/GstPulseSrc:pulsesrc0
Error: gst-stream-error-quark: Internal data stream error. (1) ../gstreamer/libs/gst/base/gstbasesrc.c(3127): gst_base_src_loop (): /GstPipeline:pipeline0/GstPulseSrc:pulsesrc0:
streaming stopped, reason not-negotiated (-4)
Traceback (most recent call last):
  File "/usr/local/share/mousai/mousai/backend/voice_recorder.py", line 94, in _on_gst_message
    self.props.peak = message.get_structure().get_value('peak')[0]
IndexError: list index out of range
```

**Comment by SeaDve** [comment]
to explain why it is happening, mousai gets the 3rd word of 13th line of the output of pactl info. so in certain locales, it will have different number of words and yields unexpected results

**Comment by SeaDve** [comment]
ill push a fix, thanks for the help

**Comment by SeaDve** [comment]
u can set `LANGUAGE=en_US` as an environment variable when running mousai so you dont have to set the global locale of your system

**Comment by KpauMupa** [comment]
> > that's really weird, may you show the output of `pactl info`? it may be helpful
> 
> So I looked thru the list of packages of GNOME and installed `soundtouch`, rebooted - now Mousai works...
> 
> I'm fairly new to linux and I don't know what this package does, but it worked out for me.
> BTW, here's my `pactl info` _after_ I installed soundtouch:
> 
> ```
> Server String: /run/user/1000/pulse/native
> Library Protocol Version: 35
> Server Protocol Version: 35
> Is Local: yes
> Client Index: 6
> Tile Size: 65472
> User Name: anonymous
> Host Name: MyCompooter
> Server Name: pulseaudio
> Server Version: 15.0
> Default Sample Specification: s16le 2ch 44100Hz
> Default Channel Map: front-left,front-right
> Default Sink: alsa_output.pci-0000_29_00.6.analog-stereo
> Default Source: alsa_output.pci-0000_29_00.6.analog-stereo.monitor
> Cookie: 9aa9:ebc6```
> ```

And yes, here my locale was still changed to english  =)
Just to clarify.

**Comment by KpauMupa** [comment]
> ill push a fix, thanks for the help

Could you please notify me (e.g. close this issue or just ping me here) when this issue is fixed in the AUR package?

**Comment by SeaDve** [comment]
im not the maintainer there, so not sure, but it should come a few hours after i do release on flathub

**Comment by SeaDve** [comment]
just published the hopeful fix

**Comment by SeaDve** [comment]
@KpauMupa the fix seems to be pushed to AUR see version `0.6.6`

**Comment by KpauMupa** [comment]
Mousai 0.6.6-1 from the AUR - the issue with ru_RU.UTF-8 locale is still there;
compiled Mousai from source (this repo) - everything's fine =)
